app/src/org/commcare/activities/connect/ConnectIdPhoneActivity.java

```diff
@@ -90,33 +86,19 @@ void displayNumber(String fullNumber) {
         if (fullNumber != null && fullNumber.startsWith("+" + codeText)) {
             fullNumber = fullNumber.substring(codeText.length() + 1);
         }
-
+        skipPhoneNumberCheck = false;
         uiController.setPhoneNumber(fullNumber);
+        skipPhoneNumberCheck = true;
```

Comment:
Should the line below (setCountryCode) be moved up above line 89? Thinking we want to make sure that both the number and country code have been updated to the latest selection before letting the API call trigger

Reply:
checked this on debug